components/core/src/LogTypeDictionaryEntry.cpp

```diff
@@ -23,7 +23,10 @@ static void escape_variable_delimiters (const string& value, size_t begin_ix, si
         auto c = value[i];
 
         // Add escape character if necessary
-        if ((char)LogTypeDictionaryEntry::VarDelim::NonDouble == c || (char)LogTypeDictionaryEntry::VarDelim::Double == c || cEscapeChar == c) {
+        if (enum_to_underlying_type(LogTypeDictionaryEntry::VarDelim::Integer) == c ||
```

Comment:
If you use `enum_to_underlying_type`, you should set the type for the enum class to `char`: `enum class VarDelim : char {`